diff -urNp -x '*.orig' ghasher-1.2.1.org/Makefile ghasher-1.2.1/Makefile
--- ghasher-1.2.1.org/Makefile	2005-11-22 18:24:41.000000000 +0100
+++ ghasher-1.2.1/Makefile	2021-10-07 00:05:38.142266834 +0200
@@ -2,9 +2,10 @@ CFLAGS = -Wall  -ggdb -D_FILE_OFFSET_BIT
 CFLAGS +=`pkg-config gtk+-2.0 libglade-2.0 --cflags`
 LDFLAGS =`pkg-config gtk+-2.0 libglade-2.0 --libs` -lcrypto -Wl,-O2
 OBJECTS = gui.o util.o list.o verify.o
-BINDIR=/usr/bin
-PIXMAPDIR=/usr/share/pixmaps
-DESKTOPFILEDIR=/usr/share/applications/
+DESTDIR=
+BINDIR=${DESTDIR}/usr/bin
+PIXMAPDIR=${DESTDIR}/usr/share/pixmaps
+DESKTOPFILEDIR=${DESTDIR}/usr/share/applications/
 
 ghasher: ${OBJECTS} 
 	${CC} -o ghasher ${OBJECTS} ${LDFLAGS} 
@@ -24,9 +25,9 @@ verify.o: ghasher.h
 .PHONY: install clean
 
 install: ghasher
-	install -c -o root ./ghasher ${BINDIR}/ghasher
-	install -c -m 444 -o root ghasher.desktop ${DESKTOPFILEDIR}/utelsystems-ghasher.desktop
-	install -c -m 444 -o root hash.xpm ${PIXMAPDIR}/hash.xpm
+	install -D ./ghasher ${BINDIR}/ghasher
+	install -D -m 444 ghasher.desktop ${DESKTOPFILEDIR}/ghasher.desktop
+	install -D -m 444 hash.xpm ${PIXMAPDIR}/ghasher.xpm
 
 clean:
 	rm -f  *.o ghasher ghasher-glade.c text2cstring
