diff -urNp -x '*.orig' ghasher-1.2.1.org/ghasher.desktop ghasher-1.2.1/ghasher.desktop
--- ghasher-1.2.1.org/ghasher.desktop	2005-11-22 18:04:54.000000000 +0100
+++ ghasher-1.2.1/ghasher.desktop	2021-10-07 00:05:38.192266666 +0200
@@ -1,10 +1,14 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=MD5 sum utility
-_Comment=Calculate the md5 sum of a file
+Name=ghasher
+GenericName=MD5 sum utility
+GenericName[pl]=Narzędzie do obliczania sum MD5
+Comment=Calculate the md5 sum of a file
+Comment[pl]=Oblicza sumę md5 pliku
 Exec=ghasher %F
 Terminal=false
 Type=Application
-Icon=hash.xpm
-Categories=Utility;GTK;X-Fedora;
+Icon=ghasher.xpm
+Categories=Utility;GTK;
+# vi: encoding=utf-8
 
